package.json

```diff
@@ -11,7 +11,11 @@
     "clear": "docusaurus clear",
     "serve": "docusaurus serve",
     "write-translations": "docusaurus write-translations",
-    "write-heading-ids": "docusaurus write-heading-ids"
+    "write-heading-ids": "docusaurus write-heading-ids",
+    "prepare": "husky install"
+  },
+  "lint-staged": {
+    "**/*": "prettier --write --ignore-unknown"
```

Comment:
> What effect does the `--ignore-unknown` have here? As far as I understand, it would ignore files that are not matched by the pattern. But since we use `**/*` any file would be catched and nothing would be ignored, correct?

It iterates through all the files in the project, ignores unsupported file types, and formats supported files to the specified style, then writes the changes back to the source file.



> That being said: Is the point of this PR to catch inconsistencies in `*.md` files only? If so, maybe we can adjust the pattern accordingly.

No, it's not just `*.md` files that are supported, since the Docusaurus framework is used, if you make changes to the source code of a document it will also be prettier formatted.

Reply:
> formats supported files

Where can I see which files are supported?

> if you make changes to the source code of a document it will also be prettier formatted

The docs are all `*.md` files as far I can tell. The docusaurus source files are quite limited in quantity, as you can see if you take a look at the `src` folder:

<img width="207" alt="image" src="https://github.com/w3f/Grants-Program/assets/5393704/cb97bdaa-2958-4bd1-a07d-34982e378cf4">

I think the emphasis should lie on linting the `*.md` files since the vast majority of changes will be made there, rather than in js files. Something particularly useful would be to check whether the `*.md` files contain any invalid `html` code since that can cause Docusaurus to break.

Reply:
> Where can I see which files are supported?
As far as I know, `.txt` files will not have its effect, if you need to have its effect on `.md` files only, we can change it to:

```json
"lint-staged": {
  "docs/**/*.md": "prettier --write",
  "applications/**/*.md": "prettier --write"
}
```